resources/top_level_overview.markdown

```diff
@@ -0,0 +1,283 @@
+
+
+#### Steps in a MapReduce Process:
+
+The following is a simplification of the steps outlined in
+_Map Reduce: Simplified Data Processing on Large Clusters_.  After we break down
+the process, we'll explore how Hadoop handles each of the following:
+
+1. The MapReduce library splits the input into _M_ chunks, and starts up copies
+of the program on multiple machines.
+
+2. The _master_ fork assigns tasks to the _worker_ nodes who are just running
+copies of the _user program_.  There are `M` map tasks, and `R` reduce tasks
+that the _master_ must assign. Note that a worker node takes on the role of
+either map, or reduce. Typically, there are many more map nodes than reduce
+nodes.
+
+3. A _worker_ assigned a map task, reads in the contents of the corresponding
+input split.  The _worker_ parses (key, values) pairs and passes it to the map
+function. The _intermediate_ (key, value) pairs produced by the mapping are
+buffered into memory.
+
+4. The local disk it partitioned into `R` regions by the _partitioning
+function_. The buffered _intermediate_ pairs produced by a map worker are
+periodically written to these partitions, with the particular location of these
+writes sent back to `master` for use by the reduce workers.
+
+5. When the `master` notifies a _reduce_ worker about a particular location of
+_intermediate_ key-value pairs, it reads the buffered data from the local
+disks of the map workers. When a reduce worker has read all intermediate data,
+it sorts it by key so that all of the same keys are grouped together. The
+sorting is needed because many different keys can map to the same reduce task.
+If the amount of intermediate data is too large to fit in memory, an external
+sort is used.
+
+6. The reduce worker iterates over the sorted data and for each unique
+intermediate key encountered, it passes the key and the corresponding set of
+intermediate values to the user’s _reduce_ function. The output of the reduce
+function is appended to a final output file for this reduce partition.
+
+7. When all map tasks and reduce tasks have been completed, the call to
+MapReduce reaches a return statement back to user code.
+
+## How Hadoop Handles MapReduce
+Now that we are comfortable with what MapReduce is, let's look into
+understanding how Hadoop handles MapReduce.  Here, each point expands on the
+corresponding number in [Steps in a MapReduce Process](#steps-in-a-map-reduce-process).
+
+Before a job can be started, however, it must run through a few steps:
+1. A job must be submitted via the user API.
+2. The resource manager retrieves the job ID (application ID)
+3. The job client checks output specification, computes splits (which can also
+  be changed to be computed over the cluster, useful for a job with many splits),
+  and then copy job resources over to the HDFS.
+
+#### Point 1 & 2: Splitting and Assigning Tasks
+HDFS (Hadoop Distributed File System) breaks down large files into configurable
```

Comment:
"breaks down" is a bit weird. Perhaps the action should be "to split" and the thing should be "a block". Sorry to be annoying with terminology, but being consistent, even it we say the same word over and over, is always a good idea.

Reply:
Not at all! I'll comb through and make sure there is consistency throughout.